composer/datasets/in_context_learning_evaluation.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import textwrap
+
+import torch
+from datasets import load_dataset
+from torch.utils.data import DataLoader, Dataset
+from transformers import AutoTokenizer
+
+from composer.core import DataSpec
+from composer.utils import dist, ensure_tuple
+from composer.utils.file_helpers import get_file
+
+
+class InContextLearningLMTaskDataset(Dataset):
+
+    def __init__(
+        self,
+        dataset_uri: str,
+        tokenizer: AutoTokenizer,
+        max_seq_len: int,
+        eos_tok_id: int,
+        destination_path: str = 'icl_lm_task.json',
+    ):
+        get_file(dataset_uri, destination_path, overwrite=True)
```

Comment:
I think it's probably safer to overwrite just in case the existing file contains data for a different eval task (even thought right now I only designed this w/ LAMBADA in mind)